packages/connectkit/src/components/Pages/WaitingDepositAddress/index.tsx

```diff
@@ -39,32 +42,30 @@ const WaitingBitcoin: React.FC = () => {
     return () => clearInterval(interval);
   }, [daimoPayOrder?.id]);
 
-  const [option, setOption] = useState<{
-    address: string;
-    amount: number;
-    uri: string;
-  }>();
+  const [details, setDetails] = useState<DepositAddressPaymentOptionData>();
   const [failed, setFailed] = useState(false);
 
   useEffect(() => {
-    payWithBitcoin().then((option) => {
-      if (!option) setFailed(true);
-      else setOption(option);
+    if (!selectedDepositAddressOption) return;
+
+    payWithDepositAddress(selectedDepositAddressOption.id).then((details) => {
+      if (!details) setFailed(true);
+      else setDetails(details);
     });
-  }, []);
+  }, [selectedDepositAddressOption]);
 
   useEffect(() => {
     triggerResize();
-  }, [option]);
+  }, [details]);
 
   return (
     <PageContent>
       {failed ? (
         <ModalContent style={{ marginLeft: 24, marginRight: 24 }}>
-          <ModalH1>Bitcoin unavailable</ModalH1>
+          <ModalH1>{selectedDepositAddressOption} unavailable</ModalH1>
```

Comment:
Further down you have `selectedDepositAddressOption.id` -> is this an object / can you use it as string here?

Maybe make a `chainName` var that is "Bitcoin" / "Tron" / etc